File: data-structures/dictionaries/visual-basic/results_dictionary.vb
' Raspberry Pi Foundation
' Usage licensed under CC BY-SA 4
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program


Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Dim results As Dictionary(Of String, Integer) = New Dictionary(Of String, Integer)() From {
        {"Detra", 17},
        {"Nova", 84},
        {"Charlie", 22},
        {"Hwa", 75},
        {"Roxann", 92},
        {"Elsa", 29}}
        Dim score As Integer = results("Charlie")
        Console.WriteLine($"Charlie's score is {score}")
        
        'Add new entry to dictionary
        results("Bob") = 78
        Console.WriteLine($"Bob has been added with score {results("Bob")}")

        'Update an entry in the dictionary
        results("Hwa") = 71
        Console.WriteLine($"Hwa's new score is {results("Hwa")}")

        'Delete an entry from the dictionary
        results.Remove("Elsa")
        Console.WriteLine("Elsa has been deleted" & vbLf)
        
        'Display the entire dictionary
        For Each kvp As KeyValuePair(Of String, Integer) In results
            Dim key As String = kvp.Key
            Dim value As Integer = kvp.Value
            Console.WriteLine($"Key: {key}  Value: {value}")
        Next
    
    End Sub


End Module

File: data-structures/dictionaries/java/results_dictionary.java
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science,
part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.util.HashMap;
import java.util.Map;

class Dictionaries
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args)
    {
        // Initialise a new dictionary of students and results
        Map<String, Integer> results = new HashMap<String, Integer>();
        results.put("Detra", 17);
        results.put("Nova", 84);
        results.put("Charlie", 22);
        results.put("Hwa", 75);
        results.put("Roxann", 92);
        results.put("Elsa", 29);

        // Display the score of Charlie
        int score = results.get("Charlie");
        System.out.println("Charlie's score is " + score + "\n");

        // Insert a new key-value pair for Bob
        results.put("Bob", 78);
        System.out.println("Bob has been added with score " + results.get("Bob") + "\n");

        // Update the score of Hwa
        results.put("Hwa", 71);
        System.out.println("Hwa's new score is " + results.get("Hwa") + "\n");

        // Delete the key-value pair for Elsa
        results.remove("Elsa");
        System.out.println("Elsa has been deleted\n");

        // Repeat for each key-value pair in the dictionary
        for (String key : results.keySet()) {
            // Output the key-value pair of each item
            int value = results.get(key);
            System.out.println("Key: " + key + "  Value: " + value);
        }
    }


}